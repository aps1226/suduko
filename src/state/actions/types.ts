export const SET_BOARD: string = 'SET_BOARD';

export const CHANGE_COLOR: string = 'CHANGE_COLOR';

export const SET_SELECTION: string = 'SET_SELECTION';