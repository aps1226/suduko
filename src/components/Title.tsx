import React,{ useEffect } from "react";
import { ImageBackground, StyleSheet, Text, View } from "react-native";
import TextStroke from './TextStroke';

export default function Title() {

  return (
    <View style = {styles.container}>
      <TextStroke stroke = {2} color = {'white'}>
        <Text style = {styles.winnerText}>
          Suduko
        </Text>
      </TextStroke>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection:'row',
    position:'absolute',
    top:'30%'
  },
  winnerText:{
    fontSize:80,
    color:'#1E90FF',
    fontWeight: 'bold',
    letterSpacing: 0.25,
    elevation:3,
  }
});