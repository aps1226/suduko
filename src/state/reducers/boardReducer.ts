import { AnyAction } from 'redux';
import { SET_BOARD } from '../actions/types';
import { shuffle } from '../boardController';

const initBoard:(number | null)[][] = [shuffle([1,2,3,4,5,6,7,8,9]),[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null]];

const boardReducer = (state = initBoard, action: AnyAction):(number | null)[][]  =>{
  switch(action.type){
    case SET_BOARD:
      return action.payload;
    default:
      return state;
  }
}

export default boardReducer;