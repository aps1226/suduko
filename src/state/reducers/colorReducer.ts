import {CHANGE_COLOR,DEFAULT_COLORS} from '../actions/types';

import {Colors} from '../../../types'

const initGradient = ():string[] =>{
  const woodPalate:string[] = ["#a37e5c", "#b89372","#c8a484","#dcb999", "#d3af8f","#d9b696","#ceaa89","#d2ae8d","#b38e6d","aa8563"];
  const gradient:string[] = [];
  for(let i = 0; i < 100; i++){
    gradient.push(woodPalate[Math.round(Math.random()*10)])
  }
  return gradient;
}

const initializedGradient:string[] = initGradient();

const initialState:Colors = {
  1:['transparent','black'],
  2:['transparent','black'],
  3:['transparent','black'],
  4:['transparent','black'],
  5:['transparent','black'],
  6:['transparent','black'],
  7:['transparent','black'],
  8:['transparent','black'],
  9:['transparent','black'],
  'Selection_1':'transparent',
  'Selection_2':'transparent',
  'Selection_3':'transparent',
  'Selection_4':'transparent',
  'Selection_5':'transparent',
  'Selection_6':'transparent',
  'Selection_7':'transparent',
  'Selection_8':'transparent',
  'Selection_9':'transparent',
  'gradient': initializedGradient
}
const colorReducer = (state = initialState, action:any) =>{

  const initialState:Colors = {
    1:['transparent','black'],
    2:['transparent','black'],
    3:['transparent','black'],
    4:['transparent','black'],
    5:['transparent','black'],
    6:['transparent','black'],
    7:['transparent','black'],
    8:['transparent','black'],
    9:['transparent','black'],
    'Selection_1':'transparent',
    'Selection_2':'transparent',
    'Selection_3':'transparent',
    'Selection_4':'transparent',
    'Selection_5':'transparent',
    'Selection_6':'transparent',
    'Selection_7':'transparent',
    'Selection_8':'transparent',
    'Selection_9':'transparent',
    'gradient': initializedGradient
  }
  
  switch(action.type){
    case CHANGE_COLOR:
      if(!action.payload.includes('Selection')) initialState[action.payload] = ['#008000','#008000'];
      else initialState[action.payload] = '#3297FD';
      return initialState;
    case DEFAULT_COLORS:
      return initialState;
    default:
      return state;
  }
}

export default colorReducer;