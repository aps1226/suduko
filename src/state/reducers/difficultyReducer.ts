import { SET_DIFFICULTY } from '../actions/types';
const difficultyReducer = (state = null, action: any) =>{
  switch(action.type){
    case SET_DIFFICULTY:
      return action.payload;
    default:
      return state;
  }
}

export default difficultyReducer;