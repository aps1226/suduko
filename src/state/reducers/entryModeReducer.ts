import {SET_ENTRY_MODE} from '../actions/types';

const entryModeReducer = (state = true, action: any) =>{
  switch(action.type){
    case SET_ENTRY_MODE:
      return action.payload;
    default:
      return state;
  }
}

export default entryModeReducer;