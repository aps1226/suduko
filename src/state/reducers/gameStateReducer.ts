import { AnyAction } from 'redux';
import { SET_GAME_STATE } from '../actions/types';
import { GameState } from '../../../types';

const initGameState:GameState = {
  'isCompleted':false,
  'gameExists':false
}
const gameStateReducer = (state = initGameState, action: AnyAction):GameState =>{
  switch(action.type){
    case SET_GAME_STATE:
      return action.payload;
    default:
      return state;
  }
}

export default gameStateReducer;