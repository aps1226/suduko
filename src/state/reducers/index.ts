import {combineReducers} from 'redux';
import boardReducer from './boardReducer';
import colorReducer from './colorReducer';
import selectionReducer from './selectionReducer';

const reducers = combineReducers({
  board: boardReducer,
  colors: colorReducer,
  selection:selectionReducer
});

export default reducers;