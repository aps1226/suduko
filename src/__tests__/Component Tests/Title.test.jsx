import React from 'react';
import { Text,View } from "react-native";
import { fireEvent, render, waitFor} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import {GameState} from '../../../types';
import {RootState} from '../../state/reducers/index';

import Title from '../../../components/Title';
//Title component test.
describe('Title', () =>{
  it('renders without crashing', () => {
    const titleComponent = render(<Title/>).toJSON();
    expect(titleComponent).toMatchSnapshot();
  });
});