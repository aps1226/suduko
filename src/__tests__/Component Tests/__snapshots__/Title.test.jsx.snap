// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders without crashing 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "position": "absolute",
      "top": "30%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": 0,
              "width": -2.4,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": 0,
              "width": 2,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": -2.4,
              "width": 0,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": -2,
              "width": -2,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": -2,
              "width": 2,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1E90FF",
            "elevation": 3,
            "fontSize": 80,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "textShadowColor": "white",
            "textShadowOffset": Object {
              "height": 2,
              "width": -2,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Suduko
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#1E90FF",
          "elevation": 3,
          "fontSize": 80,
          "fontWeight": "bold",
          "letterSpacing": 0.25,
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
</View>
`;
